packages/nvidia_nat_nemo_customizer/src/nat/plugins/customizer/dpo/trainer_adapter.py

```
@@ -0,0 +1,550 @@
+        # Create dataset repo in datastore
+        self.hf_api.create_repo(repo_id, repo_type="dataset", exist_ok=True)
+
+        # Register dataset in entity store
+        try:
+            self.entity_client.datasets.create(
+                name=dataset_name,
+                namespace=self.adapter_config.namespace,
+                files_url=f"hf://datasets/{repo_id}",
+                description=f"NAT DPO training dataset for run {run_id}",
+            )
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)